File: data/readable_format/cpp/file1700.cpp
#include <iostream>

void ugetio(int& iopt, int& nin, int& nout) {
    static int nind = 1;
    static int noutd = 2;

    if (iopt == 3) {
        noutd = nout;
    } else if (iopt == 2) {
        nind = nin;
    } else if (iopt == 1) {
        nin = nind;
        nout = noutd;
    }
}

int main() {
    int nin, nout;

    int iopt1 = 1;
    // Test iopt = 1
    ugetio(iopt1, nin, nout);
    std::cout << "Test iopt=1: nin=" << nin << " nout=" << nout << std::endl;

    int iopt2 = 2;
    nin = 10;
    // Test iopt = 2
    ugetio(iopt2, nin, nout);
    std::cout << "Test iopt=2: nin=" << nin << " nout=" << nout << std::endl;

    int iopt3 = 3;
    nout = 20;
    // Test iopt = 3
    ugetio(iopt3, nin, nout);
    std::cout << "Test iopt=3: nin=" << nin << " nout=" << nout << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1700.f90
subroutine ugetio(iopt,nin,nout)
    integer :: iopt,nin,nout
    integer, save :: nind=1, noutd=2
    if (iopt == 3) then
        noutd = nout
    else if (iopt == 2) then
        nind = nin
    else if (iopt == 1) then
        nin = nind
        nout = noutd
    end if
end subroutine ugetio

program test_ugetio
    implicit none
    integer :: nin, nout

    ! Test iopt = 1
    call ugetio(1, nin, nout)
    print *, 'Test iopt=1: nin=', nin, 'nout=', nout

    ! Test iopt = 2
    nin = 10
    call ugetio(2, nin, nout)
    print *, 'Test iopt=2: nin=', nin, 'nout=', nout

    ! Test iopt = 3
    nout = 20
    call ugetio(3, nin, nout)
    print *, 'Test iopt=3: nin=', nin, 'nout=', nout

end program test_ugetio